File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int,int> freq;
        for(auto &num : nums)freq[num]++;

        int ans = 0;
        for(auto &pair : freq){
            int num = pair.first;
            if(freq.count(num+1))
                ans = max(ans,freq[num] + freq[num + 1]);
        }
        return ans;
    }
};

File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.cs
public class Solution {
    public int FindLHS(int[] nums) {
        Dictionary<int,int> freq = new Dictionary<int,int>();
        foreach(int num in nums){
            if(freq.ContainsKey(num))freq[num]++;
            else freq[num] = 1;
        }

        int ans = 0;
        foreach(var pair in freq){
            int num = pair.Key;
            if(freq.ContainsKey(num + 1))
                ans = Math.Max(ans,freq[num] + freq[num + 1]);
        }
        return ans;
    }
}